ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/location/CaptureLocationTaskMapFragment.kt

```diff
@@ -46,18 +46,27 @@ class CaptureLocationTaskMapFragment(private val viewModel: CaptureLocationTaskV
   ): View {
     val root = super.onCreateView(inflater, container, savedInstanceState)
     viewLifecycleOwner.lifecycleScope.launch {
-      getMapViewModel().getLocationUpdates().collect { viewModel.updateLocation(it) }
+      if (this@CaptureLocationTaskMapFragment::viewModel.isInitialized) {
+        getMapViewModel().getLocationUpdates().collect { viewModel.updateLocation(it) }
+      }
     }
     return root
   }
 
   override fun onMapReady(map: MapFragment) {
     binding.locationLockBtn.isClickable = false
-    viewLifecycleOwner.lifecycleScope.launch { viewModel.onMapReady(mapViewModel) }
+    viewLifecycleOwner.lifecycleScope.launch {
+      if (this@CaptureLocationTaskMapFragment::viewModel.isInitialized) {
+        viewModel.onMapReady(mapViewModel)
+      }
+    }
+  }
+
+  fun setViewModel(viewModel: CaptureLocationTaskViewModel) {
```

Comment:
Might I know why don't we use `private val viewModel: CaptureLocationTaskViewModel by viewModels()`, but use the setter instead?

Reply:
because we don't want another instance of it, and need to access the one which was there in previous fragment, and  the other side, if we try accessing it using the parentFragmentManager, the parent is DataCollectionFragment not the CaptureLocationTaskFragment,

Any other approach if you could here?

Reply:
Also, it will throw error

As an example
```
java.lang.RuntimeException: Cannot create an instance of class com.google.android.ground.ui.datacollection.tasks.polygon.DrawAreaTaskViewModel
```

Reply:
The ViewModel instance is created by `AbstractTaskFragment`